scripts/ci/openshift.sh

```diff
@@ -25,28 +26,43 @@ scale_worker_nodes() {
     if [[ -n "${ARTIFACT_DIR:-}" ]]; then
         local scale_debug_dir="${ARTIFACT_DIR}/openshift/scaling-debug"
         mkdir -p "${scale_debug_dir}"
-        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/machineset.json"
-        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/machines.json"
+        oc -n openshift-machine-api get machineset -o json > "${scale_debug_dir}/initial_machineset.json"
+        oc -n openshift-machine-api get machines -o json > "${scale_debug_dir}/initial_machines.json"
     fi
 
-    original_count="$(get_running_worker_count)"
+    # note: timeout of this loop is handled by the calling context
+    while [[ "$(get_running_worker_count)" != "$expected_initial_worker_count" ]]; do
```

Comment:
Maybe we actually put the max attempt count here and bump the increment by the delta of missing workers so that we get (3) you suggested [here](https://redhat-internal.slack.com/archives/C04Q0KQMEUV/p1683819102633519?thread_ts=1683818333.352239&cid=C04Q0KQMEUV). WDYT?

Reply:
I'd prefer to do that as a separate PR.